internal/internal_workflow_client.go

```diff
@@ -1057,6 +1057,48 @@ func (wc *WorkflowClient) GetWorkerTaskReachability(ctx context.Context, options
 	return converted, nil
 }
 
+// GetWorkflowExecutionOptions returns the current WorkflowExecutionOptions of an existing workflow.
+// NOTE: Experimental
+func (wc *WorkflowClient) GetWorkflowExecutionOptions(ctx context.Context, request GetWorkflowExecutionOptionsRequest) (WorkflowExecutionOptions, error) {
+	// no side-effects, i.e., GET, when UpdateMask is empty and WorkflowExecutionOptions is not nil
+	return wc.UpdateWorkflowExecutionOptions(ctx, UpdateWorkflowExecutionOptionsRequest{
+		WorkflowId:               request.WorkflowId,
+		RunId:                    request.RunId,
+		WorkflowExecutionOptions: WorkflowExecutionOptions{},
+		UpdatedFields:            []string{},
+	})
+}
+
+// UpdateWorkflowExecutionOptions partially overrides the WorkflowExecutionOptions of an existing workflow execution,
+// and returns the new WorkflowExecutionOptions after applying the changes.
+// It is intended for building tools that can selectively apply ad-hoc workflow configuration changes.
+// NOTE: Experimental
+func (wc *WorkflowClient) UpdateWorkflowExecutionOptions(ctx context.Context, request UpdateWorkflowExecutionOptionsRequest) (WorkflowExecutionOptions, error) {
```

Comment:
Calling this a "request" is unique amongst our high-level client calls (i.e. ones that don't just forward the protos). I understand the apprehension for having `OptionsOptions`, but we are the ones that chose to use the word "options" in the API when the "properties" event was available in history. I don't have a strong opinion here, it's just a bit confusing to have a new, inconsistent concept (using a "request" vs "options") for naming reasons.

Reply:
I was coping what we did for `QueryWorkflowWithOptionsRequest` but happy to call it to something else. Definetly not OptionsOptions, any suggestions?

Reply:
Yeah, we are unfortunately inconsistent here (same with whether this is a pointer or not, though I think it should remain not a pointer). I am fine with what you have here as a request, though would like to wait for @Quinn-With-Two-Ns's opinion too.

Reply:
I need to move on, trying to make a dependent PR today, but we can revisit naming before merging into main...